katdal/visdatav4.py

```diff
@@ -101,11 +103,15 @@ class VisibilityDataV4(DataSet):
         (default is first antenna in use)
     time_offset : float, optional
         Offset to add to all correlator timestamps, in seconds
+    applycal : string, optional
+        Comma-separated calibration products to apply to vis/weights/flags
+        (default is no calibration)
```

Comment:
I'm mindful that someone will probably pass this in from outside katdal, via a `katdal.open` keyword argument or maybe even a URL query string. I considered both options and are open for discussion on the relative merits of each :-) I should probably accept both anyway like `select()` does.

Also, instead of defining a constant that needs to be imported before use, one could simply have a special string (e.g. "all" or "L1") to get a specific combination of cal products. This matches the behaviour of e.g. flag selection in `DataSet.select`.

Reply:
Fixed to accept both forms of product lists, and added a note to the docstring about the future of `applycal`.